### PR TITLE
Improve test coverage for database package

### LINKED ISSUE #1296: increase test coverage in pkg/database
especially around data validation, i.e. `BeforeSave` methods

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes #1296

**Release Note**

```release-note
Improve database test coverage
```

### CODE REVIEW

File: pkg/database/authorized_app.go [hunk 1]
```diff
@@ -109,51 +109,20 @@ func (a *AuthorizedApp) IsDeviceType() bool {
...
-// TableName definition for the authorized apps relation.
-func (AuthorizedApp) TableName() string {
-	return "authorized_apps"
-}
-
-// CreateAuthorizedApp generates a new API key and assigns it to the specified
-// app. Note that the API key is NOT stored in the database, only a hash. The
-// only time the API key is available is as the string return parameter from
-// invoking this function.
-func (r *Realm) CreateAuthorizedApp(db *Database, app *AuthorizedApp, actor Auditable) (string, error) {
```
Comment (sethvargo) [inline pkg/database/authorized_app.go:130]: Moved into Realm

File: pkg/database/cleanup.go [hunk 2]
```diff
@@ -34,25 +34,22 @@ type CleanupStatus struct {
...
 }
 
-// TableName sets the CleanupStatus table name
-func (CleanupStatus) TableName() string {
-	return "cleanup_statuses"
-}
-
 // CreateCleanup is used to create a new 'cleanup' type/row in the database.
 func (db *Database) CreateCleanup(cType string) (*CleanupStatus, error) {
-	cstat := &CleanupStatus{
```
Comment (sethvargo) [inline pkg/database/cleanup.go:44]: 🐉 t'was a sad bug here

File: pkg/database/signing_key.go [hunk 3]
```diff
@@ -38,11 +38,3 @@ type SigningKey struct {
 func (s *SigningKey) GetKID() string {
 	return fmt.Sprintf("r%dv%d", s.RealmID, s.ID)
 }
-
-func (s *SigningKey) Delete(db *Database) error {
```
Comment (sethvargo) [inline pkg/database/signing_key.go:42]: These are both unused

**Review by whaught — APPROVED** [review]

File: pkg/database/audit_entry.go [hunk 4]
```diff
@@ -63,6 +66,32 @@ type AuditEntry struct {
 	CreatedAt time.Time
 }
 
+// BeforeSave runs validations. If there are errors, the save fails.
+func (a *AuditEntry) BeforeSave(tx *gorm.DB) error {
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply pkg/database/audit_entry.go:70]: These fields are NOT NULL in the database. A runtime error would already have existed, just with a pgx_violated_contstraint response instead.

**Pull request merged**